lib/clAdapters/google-mail/index.js

```diff
@@ -229,21 +271,18 @@ var convertEmailListToArrayOfEmailAddressObjects = function(emailList) {
 };
 
 var hasLabel = function(message, labelValue) {
-  return message.labelIds.indexOf(labelValue) >= 0;
+  return message.labelIds && message.labelIds.length && (message.labelIds.indexOf(labelValue) >= 0);
 };
 
 var mapEmailData = function(emailData) {
   var mappedData = [];
 
   for (var userIter = 0; userIter < emailData.length; userIter++) {
```

Comment:
when iterating over arrays like this and calling a function in the body, there is not much performance gain over using `_.map` or `[].map`, and it would be a lot clearer (reduce the need for separate index variables / repeated access into the array).